rh/schema.py

```diff
@@ -26,6 +37,46 @@ def get_slack_notify_text(rhr: models.RentalHistoryRequest) -> str:
     return f"{user_text} has requested {rh_link}!"
 
 
+def run_rent_stab_sql_query(bbl: str) -> Optional[Dict[str, Any]]:
+    sql_query = """
+        select uc2007, uc2008, uc2009, uc2010, uc2011, uc2012, uc2013,
+             uc2014, uc2015, uc2016, uc2017, uc2018, uc2019
+        from rentstab
+        full join rentstab_v2 using(ucbbl)
+        where ucbbl = %(bbl)s
+    """
+
+    with connections[settings.NYCDB_DATABASE].cursor() as cursor:
+        cursor.execute(sql_query, {"bbl": bbl})
+        json_result = list(generate_json_rows(cursor))
+        if not json_result:
+            return None
+        return json_result[0]
+
+
+def process_rent_stab_data(raw_data: Dict[str, Any]) -> Dict[str, Any]:
+    for item in sorted(raw_data.items(), reverse=True):
+        if item[1] and item[1] > 0:
+            return {"latest_year": item[0].replace("uc", ""), "latest_unit_count": item[1]}
+    return BLANK_RENT_STAB_INFO
+
+
+def get_rent_stab_info_for_bbl(bbl: str) -> Optional[Dict[str, Any]]:
+    # Case 1: No connection to the database
+    if not settings.NYCDB_DATABASE:
+        return None
+
+    raw_data = run_rent_stab_sql_query(bbl)
+
+    # Case 2: We connected to the database, but no RS data was found
+    if not raw_data:
+        return BLANK_RENT_STAB_INFO
+
+    # Case 3: We connected to the database, and RS data was found
+    else:
+        return process_rent_stab_data(raw_data)
+
```

Comment:
New helper function here spells out our "three cases" of getting rent stab info from a bbl.

Reply:
Nice!  Based on its usage, though, I think there's a fourth case--the case where `bbl` is an empty string because the geosearch service wasn't available?  So maybe change line 66 to `if not (bbl and settings.NYCDB_DATABASE):`?

Reply:
good point!